File: ffitness/Models/Trainer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Ffitness.Models
{
    public class Trainer
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public String FirstName { get; set; }
        [Required]
        public String LastName { get; set; }
        [Required, MinLength(5)]
        public String Description { get; set; }
        public List<Activity> Activities { get; set; }
    }
}

File: ffitness/ClientApp/src/app/trainers/shared/trainer.ts
import { Activity } from "../../activities/shared/activity.model";

export namespace Trainer {

  export interface TrainerRequest {
    FirstName: string;
    LastName: string;
    Description: string;
    Activities: Number[];
    ProfilePicture?: File;
  };

  export interface TrainerResponse {
    id: number;
    firstName: string;
    lastName: string;
    description: string;
    activities: Activity[];
    profilePicture?: string;
  }

  export interface TrainerConfirmation {
    status: string;
  };
};
